experiment/resultstore/main.go

```diff
@@ -154,560 +96,70 @@ func print(inv ...interface{}) {
-	}
-	invName := "invocations/" + invID
-	inv, err = down.GetInvocation(ctx, &resultstore.GetInvocationRequest{
-		Name: invName,
-	})
-	if err != nil {
-		return fmt.Errorf("get invocation %s: %v", invName, err)
-	}
-	print("gi", tok, inv)
-
-	inv.Files = manyFiles
-	inv, err = up.UpdateInvocation(ctx, &resultstore.UpdateInvocationRequest{
-		Invocation: inv,
-		UpdateMask: &field_mask.FieldMask{
-			Paths: []string{"files"},
-		},
-		AuthorizationToken: tok,
-	})
+	result, err := download(ctx, opt)
 	if err != nil {
-		return fmt.Errorf("update invocation %s: %v", invName, err)
+		return fmt.Errorf("download %s: %v", opt.path, err)
 	}
 
-	// add a target
+	inv, target, test := convert(
+		opt.project,
+		"hello again",
+		opt.path,
+		*result,
+	)
+	print(inv.To(), test.To())
 
-	t, err := up.CreateTarget(ctx, &resultstore.CreateTargetRequest{
-		Parent:   inv.Name,
-		TargetId: "//erick//fejta:stardate-" + strconv.FormatInt(time.Now().Unix(), 10),
-		Target: &resultstore.Target{ // https://godoc.org/google.golang.org/genproto/googleapis/devtools/resultstore/v2#Target
-			StatusAttributes: &resultstore.StatusAttributes{
-				Status:      resultstore.Status_BUILDING,
-				Description: "fun fun",
-			},
-			Visible: true, // Will not appear in UI otherwise
-			//   Timing
-			//   TargetAttributes
-			//   TestAttributes
-			//   Properties
-			//   Files
-		},
-		AuthorizationToken: tok,
-	})
-	if err != nil {
-		return fmt.Errorf("create target: %v", err)
+	if opt.project == "" {
+		return nil
 	}
-	print("ct", t)
 
-	tr, err := down.ListTargets(ctx, &resultstore.ListTargetsRequest{
-		Parent: inv.Name,
-		// PageSize
-		// PageStart
-	})
+	conn, err := resultstore.Connect(ctx, opt.account)
 	if err != nil {
-		return fmt.Errorf("list targets: %v", err)
+		return fmt.Errorf("resultstore connection: %v", err)
 	}
-	print("lt", inv, tr)
-
-	// create a configuration
-	c, err := up.CreateConfiguration(ctx, &resultstore.CreateConfigurationRequest{
-		Parent:             inv.Name,
-		ConfigId:           "default",
-		AuthorizationToken: tok,
-		Configuration: &resultstore.Configuration{
-			// Overall status of this config
-			StatusAttributes: &resultstore.StatusAttributes{
-				Status:      resultstore.Status_BUILDING, // https://godoc.org/google.golang.org/genproto/googleapis/devtools/resultstore/v2#Status
-				Description: "very exciting",
-			},
-			ConfigurationAttributes: &resultstore.ConfigurationAttributes{
-				Cpu: "amd64", // this is the only value, LOL
-			},
-			Properties: []*resultstore.Property{
-				prop("something", fmt.Sprintf("exciting-%d", time.Now().Unix())),
-				prop("more", "excitement"),
-			},
-		},
-	})
-	if err != nil {
-		print("create failed (already exists?", err)
+	rsClient := resultstore.NewClient(conn).WithContext(ctx)
+	if opt.secret != "" {
+		rsClient = rsClient.WithSecret(resultstore.Secret(opt.secret))
+	} else {
+		secret := resultstore.NewSecret()
+		fmt.Println("Secret:", secret)
```

Comment:
debug line? remove?

Reply:
This is intentional. Otherwise there is no way to update the invocation after I've created it.

I typically run it once without the secret, then copy this line from the output into subsequent calls.